.github/workflows/codeql.yml

```diff
@@ -0,0 +1,74 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "master" ]
+  schedule:
+    - cron: '23 12 * * 4'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: [ubuntu-latest, windows-latest, macos-latest]
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp' ]
+        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python', 'ruby' ]
+        # Learn more about CodeQL language support at https://aka.ms/codeql-docs/language-support
```

Comment:
```suggestion
        # Learn more about CodeQL language support at https://aka.ms/codeql-docs/language-support
        os: [windows-latest, ubuntu-latest, macOS-latest]
```

Reply:
I think this won't work because the scope of `os` is above the definition of the `matrix.os`. I've just used `ubuntu-latest` for now. We can change it if something changes 😄